gef.py

```diff
@@ -10482,21 +10482,83 @@ def lookup_command(self, cmd):
 
         return None
 
+@register_command
+class AliasesCommand(GenericCommand):
+    """Base command to add, remove, or list aliases."""
 
-class GefAliases(gdb.Command):
-    """List all custom aliases."""
+    _cmdline_ = "aliases"
+    _syntax_  = "{:s} (add|rm|ls)".format(_cmdline_)
 
     def __init__(self):
-        super().__init__("aliases", gdb.COMMAND_OBSCURE, gdb.COMPLETE_NONE)
+        super().__init__(prefix=True)
         return
 
-    def invoke(self, args, from_tty):
-        self.dont_repeat()
-        ok("Aliases defined:")
-        for _alias in __aliases__:
-            gef_print("{:30s} {} {}".format(_alias._alias, RIGHT_ARROW, _alias._command))
+    def do_invoke(self, argv):
+        self.usage()
+        return
+
+@register_command
+class AliasesAddCommand(AliasesCommand):
+    """Command to add aliases."""
+
+    _cmdline_ = "aliases add"
+    _syntax_  = "{0} [ALIAS] [COMMAND]".format(_cmdline_)
+    _example_ = "{0} scope telescope".format(_cmdline_)
+
+    def __init__(self):
+        super().__init__()
+        return
+
+    def do_invoke(self, argv):
+        if (len(argv) < 2):
+            self.usage()
+            return
+        GefAlias(argv[0], " ".join(argv[1:]))
+        GefSaveCommand().invoke(None, False)
```

Comment:
Doing automatic saves is risky as it may create confusion: for example it broke the unit tests: see [here](https://github.com/hugsy/gef/pull/640/checks?check_run_id=2758217671).
The global idea is: `gef` operates in temporary mode for all settings. If the user wants to make them permanent, they would have to explicitly do a `gef save`.

Is there a reason you do automatic saves here?

Reply:
For some reason I thought the save was necessary for saving the new alias to the `~/.gef.rc` file, but testing it I can see that it isn't. Not quite sure where I came up with that 🙃